pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -90,6 +90,23 @@ public Response resumeConsumption(
     }
   }
 
+  @POST
+  @Path("/tables/{tableName}/resetConsumption")
+  @ApiOperation(value = "Reset consumption of a realtime table",
+      notes = "Reset the consumption for a realtime table using force commit")
+  public Response resetConsumption(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName) {
```

Comment:
Suggest adding an optional segment so that we only force commit one segment?

Reply:
We don't have partition level pausing. I know this endpoint is a bit different than pause/resume, but let me think about it a bit and maybe we can come up with a way for partition leve operation on all endpoints in this class.